libbeat/template/load.go

```diff
@@ -38,26 +40,33 @@ type ESClient interface {
 }
 
 type Loader struct {
-	config   TemplateConfig
-	client   ESClient
-	beatInfo beat.Info
-	fields   []byte
+	config    *TemplatesConfig
+	client    ESClient
+	beatInfo  beat.Info
+	esVersion string
 }
 
-// NewLoader creates a new template loader
-func NewLoader(cfg *common.Config, client ESClient, beatInfo beat.Info, fields []byte) (*Loader, error) {
-	config := DefaultConfig
+// NewESLoader creates a new Elasticsearch template loader
+func NewESLoader(cfg *common.Config, client ESClient, beatInfo beat.Info) (*Loader, error) {
+	return newLoader(cfg, client, beatInfo, client.GetVersion())
```

Comment:
Once you rebase onto master `client.GetVersion()` will not return a string anymore.

Reply:
Already changed that, but haven't pushed yet.